_blog/2024-02-09-using-clerk-in-rust.mdx

```diff
@@ -0,0 +1,1014 @@
+
+```
+
+You will get a few prompts to create a new project using this template. Make sure you provide a unique name because when you deploy this project on Shuttle this project name will become the sub-domain. And once a name is taken it cannot be used for other project as well:
+
+```bash
+What do you want to name your project?
+It will be hosted at ${project_name}.shuttleapp.rs, so choose something unique!
+✔ Project name · <ENTER_PROJECT_NAME>
+Where should we create this project?
+✔ Directory · <ENTER_THE_DIRECTORY_NAME>
+✔ Claim the project name <PROJECT_NAME> by starting a project container on Shuttle? <Y/N>
+
+```
+
+After that, `cd` into the project directory:
+
+```bash
+cd {project_name}
+
+```
+
+## Setup a new application in Clerk
+
+Head on to Clerk’s official website, sign in or sign up if don’t have an account, and create a new application in the dashboard.
+
+Give a name to your project and select the providers using which users can sign-in or sign-up in your application.
+
+After that, you will get a **Publishable Key** and a **Secret key** which we will use later in our application.
+
+[https://gist.github.com/assets/61370770/7fb52565-605c-42e9-9a6a-3fc7e4855696](https://gist.github.com/assets/61370770/7fb52565-605c-42e9-9a6a-3fc7e4855696)
+
+## Build the back-end
+
+Now we can start updating the existing back-end code to write and build our own endpoints for our application.
+
+`cd` into the backend:
+
+```bash
+cd backend
+
+```
+
+Copy the contents of `Secrets.example.toml` into a new `Secrets.toml` file:
+
+```bash
+touch Secrets.toml
+cp Secrets.example.toml Secrets.toml
+
+```
+
+Copy the **Secret key** from the Clerk dashboard and add it to `Secrets.toml` :
+
+```bash
+CLERK_SECRET_KEY = "sk_test_vsWrxxxxxxxxxxxxxxxxxxxxxxxxxtfTr"
+
+```
+
+> ⚠️ Do not commit this file
+> 
+
+Now, we need to write a new migration in a `schema.sql` file which will create a new `issues` table in our Postgres database.
+
+Create `schema.sql` and add the schema:
+
+```sql
+-- Create a non-existing table
+CREATE TABLE IF NOT EXISTS issues (
+  id SERIAL PRIMARY KEY,
+  title VARCHAR(100) NOT NULL,
+  description VARCHAR(300) NOT NULL,
+  status VARCHAR(10) NOT NULL,
+  label VARCHAR(10) NOT NULL,
+  author VARCHAR NOT NULL
+);
+
+```
+
+We can migrate this schema using **[SQLX](https://docs.rs/sqlx/latest/sqlx/)**. So add the SQLX crate to your dependency:
+
+```bash
+cargo add sqlx
+
+```
+
+To connect to a shared Postgres database managed by Shuttle you need to add the crate with features specific to Postgres and SQLX to your dependencies:
+
+```bash
+cargo add shuttle-shared-db -F shuttle-shared-db/postgres -F shuttle-shared-db/sqlx
+
+```
+
+> 🗨️ Even though the database is shared our data is protected and kept secured from other users by Shuttle.
+> 
+
+### Connect to the shared Postgres DB
+
+Let’s migrate this schema, `cd` into the `src` directory, and write some code in the `main.rs` file.
```

Comment:
You can simplify as "edit the ... file", since not everyone use terminal editors

Reply:
Right!